### PR TITLE
Continuation puzzle scene 1

### PR DESCRIPTION
**Pull request by AncientFungus**
An update to the first puzzle scene. After this, the only thing left to do is implementing triggers.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L583](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/04c9c64d0675ff0f1bce39220d86bbb2271f8586/src/Engine/Manager/ScriptManager.cpp#L583): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L847](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/04c9c64d0675ff0f1bce39220d86bbb2271f8586/src/Engine/Manager/ScriptManager.cpp#L847): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L853](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/04c9c64d0675ff0f1bce39220d86bbb2271f8586/src/Engine/Manager/ScriptManager.cpp#L853): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L861](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/04c9c64d0675ff0f1bce39220d86bbb2271f8586/src/Engine/Manager/ScriptManager.cpp#L861): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as [hunk 1]
```diff
@@ -2,29 +2,34 @@ class Minecart_Puzzle_1{
...
     Minecart_Puzzle_1(Entity @entity){
         @hub = Managers();
         @self = @entity;
         speed = 4.0f;
         stopTime = 0.0f;
         endTime = 7.5f;
-        hasBridgeBeenLowered = false;
         
-        trigger = false;
+        //trigger = false;
```
Comment (abdominator) [inline Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: Remove comment.

File: Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as [hunk 2]
```diff
@@ -2,29 +2,34 @@ class Minecart_Puzzle_1{
...
-            self.position.z -= speed*deltaTime;
+    void Update(float deltaTime) {
+        self.position.z -= speed*deltaTime;
         
-        if (self.GetWorldPosition().x <= 5.4)
+        if (self.GetWorldPosition().x <= 5.4 && startAgain == false)
             speed = 0.0f;
+        else if (startAgain && speed < 4.0f)
+            speed += 0.00664f;
+        else if (speed >= 4.0f) {
```
Comment (abdominator) [inline Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: Remove brackets.

File: Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as [hunk 3]
```diff
@@ -2,29 +2,34 @@ class Minecart_Puzzle_1{
...
+            speed == 4.0f;
```
Comment (abdominator) [inline Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: speed = 4.0f;

File: Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as [hunk 4]
```diff
@@ -47,8 +52,11 @@ class Minecart_Puzzle_1{
         }*/
     }
     
-    void ReceiveMessage(int signal){
-        if (signal == 1)
-            trigger = true;
+    void ReceiveMessage(Entity @sender, int signal){
+        /*if (signal == 1 && trigger == false)
```
Comment (abdominator) [inline Game/Resources/Constant Scene/Minecart/Script/Minecart_Puzzle_1.as]: Remove comments.

File: Game/Resources/NewAssets/Scripts/Bridge.as [hunk 5]
```diff
@@ -1,17 +1,32 @@
...
 
     // Called by the engine for each frame.
     void Update(float deltaTime) {
-
+        if (trigger == true && roll(self.rotation) < radians(0.0f))
+            self.RotateRoll(radians(0.25f));
+        else if (roll(self.rotation) > 0.0f) {
+            trigger = false;
+            SendMessage(cart, 3);
+            //self.rotation.x = 0.0f;
```
Comment (abdominator) [inline Game/Resources/NewAssets/Scripts/Bridge.as]: Remove comment.

File: src/Engine/Manager/ScriptManager.cpp [hunk 6]
```diff
@@ -660,9 +661,11 @@ void ScriptManager::Update(World& world, float deltaTime) {
 
                 if (script->IsInPropertyMap(name, typeId)) {
 
-                    if (typeId == engine->GetTypeIdByDecl("Entity@"))
+                    if (typeId == engine->GetTypeIdByDecl("Entity@")) {
+                        Entity* whatever = Hymn().GetEntityByGUID(*(unsigned int*)script->GetDataFromPropertyMap(name));
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MINOR Variable 'whatever' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L568](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5dc9c2b13a11d6b72ec0567c26a0a6f19b124425/src/Engine/Manager/ScriptManager.cpp#L568): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L781](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5dc9c2b13a11d6b72ec0567c26a0a6f19b124425/src/Engine/Manager/ScriptManager.cpp#L781): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L787](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5dc9c2b13a11d6b72ec0567c26a0a6f19b124425/src/Engine/Manager/ScriptManager.cpp#L787): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [TriggerObserver.cpp#L15](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5dc9c2b13a11d6b72ec0567c26a0a6f19b124425/src/Engine/Physics/TriggerObserver.cpp#L15): The function 'GetPhase' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [TriggerObserver.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/5dc9c2b13a11d6b72ec0567c26a0a6f19b124425/src/Engine/Physics/TriggerObserver.cpp#L115): The function 'addSingleResult' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L568](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/655135511ed804d87a115f8b05661f61d87461e6/src/Engine/Manager/ScriptManager.cpp#L568): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L781](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/655135511ed804d87a115f8b05661f61d87461e6/src/Engine/Manager/ScriptManager.cpp#L781): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L787](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/655135511ed804d87a115f8b05661f61d87461e6/src/Engine/Manager/ScriptManager.cpp#L787): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 7 issues
* ![MINOR][MINOR] 7 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L115](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a0e8e5f985e48a6156946d6d85e7942d5e95bbbb/src/Engine/Manager/ScriptManager.cpp#L115): Variable 'callstack' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L116](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a0e8e5f985e48a6156946d6d85e7942d5e95bbbb/src/Engine/Manager/ScriptManager.cpp#L116): Variable 'variables' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [ScriptManager.cpp#L603](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a0e8e5f985e48a6156946d6d85e7942d5e95bbbb/src/Engine/Manager/ScriptManager.cpp#L603): The function 'ClearBreakpoints' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L817](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a0e8e5f985e48a6156946d6d85e7942d5e95bbbb/src/Engine/Manager/ScriptManager.cpp#L817): The function 'GetStringDeclarationID' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L823](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/a0e8e5f985e48a6156946d6d85e7942d5e95bbbb/src/Engine/Manager/ScriptManager.cpp#L823): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SwedishSock** [comment]
Old pull request, and I don't have any conflicts on the branch when I merge master into it, so I'm closing this pull request. A better one will be made in time when puzzle scene 1 is complete.

**Pull request closed without merging**